variable/include/scipp/variable/bucket_variable.tcc

```diff
@@ -27,18 +27,20 @@ VariableConstView::constituents() const {
 }
 
 template <class T>
-std::tuple<VariableConstView, Dim, typename T::element_type>
+std::tuple<VariableView, Dim, typename T::element_type>
```

Comment:
Can you explain why this change?

Reply:
This now supports changing the bucket indices, i.e., the size of each bucket/bin. I am not sure letting clients to this is a good idea (since it is unsafe if out-of-range or overlapping indices are set), but it is also required to grow buckets, e.g., after reserving space and merging bins, see `concatenate` (to be renamed to `merge_bins` in a future PR). Since it is not exposed to Python I am at this point not too worried about the possibility to setup broken indices, but it is certainly something to keep in mind for the future, when we iterate and create a more solid interface for the bucket mechanism.